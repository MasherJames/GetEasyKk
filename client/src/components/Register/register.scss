.container {
    height: 90vh;
    display: flex;
    .register {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        width: 45vw;
        height: 80vh;
        margin: auto;
        background-color: rgb(184, 246, 253);
        .register-head {
            text-transform: uppercase;
            border-bottom: 3px solid #48BcC8;
        }
        .have-account {
            color: rgb(92, 91, 91);
            margin-top: 1rem;
        }
        .register-form {
            display: flex;
            flex-direction: column;
            .btn {
                padding: .7rem;
                background-color: #48BcC8;
                border: none;
                cursor: pointer;
                text-transform: uppercase;
                outline: none;
            }
            &>* {
                margin: .8rem;
            }
            .form-group {
                .inputs {
                    width: 30vw;
                    padding: .7rem;
                    outline: none;
                    font-size: 1rem;
                }
                .invalid-input {
                    border: 1px solid red;
                }
                .error {
                    color: rgb(255, 31, 31);
                }
            }
        }
    }
}