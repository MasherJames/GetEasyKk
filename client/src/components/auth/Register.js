import React, { Component } from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { registerRequestAction } from "../../actions/registerActions";

class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      email: "",
      password: "",
      confirmPassword: "",
      errors: {}
    };
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleSubmit = e => {
    e.preventDefault();

    const newUser = {
      email: this.state.email,
      username: this.state.username,
      password: this.state.password,
      confirmPassword: this.state.confirmPassword
    };

    this.props.registerRequestAction(newUser);
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }

  render() {
    const { errors } = this.state;
    return (
      <div>
        <div className="container">
          <div className="col-md-8 m-auto">
            <p className="lead text-center">Create Your Account</p>
            <form onSubmit={this.handleSubmit}>
              <div className="form-group">
                <input
                  type="email"
                  name="email"
                  placeholder="Email"
                  value={this.props.email}
                  onChange={this.handleChange}
                  className={classnames("form-control form-control-lg", {
                    "is-invalid": errors.email
                  })}
                />
                {errors.email && (
                  <div className="invalid-feedback">{errors.email}</div>
                )}
              </div>
              <div className="form-group">
                <input
                  type="username"
                  name="username"
                  placeholder="Username"
                  value={this.props.username}
                  onChange={this.handleChange}
                  className={classnames("form-control form-control-lg", {
                    "is-invalid": errors.username
                  })}
                />
                {errors.username && (
                  <div className="invalid-feedback">{errors.username}</div>
                )}
              </div>
              <div className="form-group">
                <input
                  type="password"
                  name="password"
                  placeholder="Password"
                  value={this.props.password}
                  onChange={this.handleChange}
                  className={classnames("form-control form-control-lg", {
                    "is-invalid": errors.password
                  })}
                />
                {errors.password && (
                  <div className="invalid-feedback">{errors.password}</div>
                )}
              </div>
              <div className="form-group">
                <input
                  type="password"
                  name="confirmPassword"
                  placeholder="confirmPassword"
                  value={this.props.confirmPassword}
                  onChange={this.handleChange}
                  className={classnames("form-control form-control-lg", {
                    "is-invalid": errors.confirmPassword
                  })}
                />
                {errors.confirmPassword && (
                  <div className="invalid-feedback">
                    {errors.confirmPassword}
                  </div>
                )}
              </div>
              <input type="submit" className="btn btn-info btn-block mt-4" />
            </form>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  errors: state.registerUser.errors,
  success: state.registerUser.success,
  payload: state.registerUser.payload,
  signingUp: state.registerUser.signingUp
});

Register.propTypes = {
  registerRequestAction: PropTypes.func.isRequired,
  signingUp: PropTypes.bool.isRequired,
  success: PropTypes.string.isRequired,
  errors: PropTypes.object.isRequired
};

export default connect(
  mapStateToProps,
  { registerRequestAction }
)(Register);
