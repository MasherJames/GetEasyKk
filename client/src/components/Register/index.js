import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { registerRequestAction } from "../../actions/registerActions";
import InputField from "../common/InputField";
import "./register.scss";

class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      email: "",
      password: "",
      confirmPassword: "",
      errors: {}
    };
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleSubmit = e => {
    e.preventDefault();

    const newUser = {
      email: this.state.email,
      username: this.state.username,
      password: this.state.password,
      confirmPassword: this.state.confirmPassword
    };

    this.props.registerRequestAction(newUser);
  };

  componentDidMount() {
    if (this.props.isAuthenticated) {
      this.props.history.push("/dashboard");
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }

  render() {
    const { errors } = this.state;
    return (
      <div className="container">
        <div className="register">
          <h2 className="register-head">Sign Up</h2>
          <Link to="/login" className="have-account">
            Already have an account ?
          </Link>
          <form onSubmit={this.handleSubmit} className="register-form">
            <InputField
              name="email"
              placeholder="Email"
              value={this.state.email}
              error={errors.email}
              type="email"
              onChange={this.handleChange}
            />
            <InputField
              name="username"
              placeholder="Username"
              value={this.state.username}
              error={errors.username}
              onChange={this.handleChange}
            />
            <InputField
              name="password"
              type="password"
              placeholder="Password"
              value={this.state.password}
              error={errors.password}
              onChange={this.handleChange}
            />
            <InputField
              name="confirmPassword"
              type="password"
              placeholder="confirmPassword"
              value={this.state.confirmPassword}
              error={errors.confirmPassword}
              onChange={this.handleChange}
            />
            <button type="submit" className="btn">
              Sign up
            </button>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  errors: state.registerUser.errors,
  success: state.registerUser.success,
  payload: state.registerUser.payload,
  signingUp: state.registerUser.signingUp,
  isAuthenticated: state.loginUser.isAuthenticated
});

Register.propTypes = {
  registerRequestAction: PropTypes.func.isRequired,
  signingUp: PropTypes.bool.isRequired,
  success: PropTypes.string.isRequired,
  errors: PropTypes.object.isRequired
};

export default connect(
  mapStateToProps,
  { registerRequestAction }
)(Register);
