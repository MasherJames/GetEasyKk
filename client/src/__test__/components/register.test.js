import React from "react";
import { shallow, mount } from "enzyme";
import Register from "../../components/Register";

describe("<Register />", () => {
  const wrapper = shallow(<Register />);
  it("renders the register component successfully", () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("shows", () => {
    expect(wrapper.find(".inputs")).toBeDefined();
  });
});
