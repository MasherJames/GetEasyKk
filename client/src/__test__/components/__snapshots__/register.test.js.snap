// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders the register component successfully 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
